dags/veda_data_pipeline/utils/collection_generation.py

```diff
@@ -0,0 +1,135 @@
+            },
+        }
+
+        # Default REQUIRED fields
+        if not collection_dict.get("title"):
+            collection_dict["description"] = dataset["collection"]
+        if not collection_dict.get("license"):
+            collection_dict["license"] = "proprietary"
+
+        return collection_dict
+
+    def _create_zarr_template(self, dataset: Dict[str, Any], store_path: str) -> dict:
+        template = self.get_template(dataset)
+        template["assets"] = {
+            "zarr": {
+                "href": store_path,
+                "title": "Zarr Array Store",
+                "description": "Zarr array store with one or several arrays (variables)",
+                "roles": ["data", "zarr"],
+                "type": "application/vnd+zarr",
+                "xarray:open_kwargs": {
+                    "engine": "zarr",
+                    "chunks": {},
+                    **dataset.xarray_kwargs,
+                },
+            }
+        }
+        return template
+
+    def create_zarr_collection(self, dataset: Dict[str, Any], role_arn: str) -> dict:
+        """
+        Creates a zarr stac collection based off of the user input
+        """
+        discovery = dataset.discovery_items[0]
+        store_path = f"s3://{discovery.bucket}/{discovery.prefix}{discovery.zarr_store}"
+        template = self._create_zarr_template(dataset, store_path)
+
+        fs = fsspec.filesystem("s3", anon=False, role_arn=role_arn)
+        store = fs.get_mapper(store_path)
+        ds = xr.open_zarr(
+            store, consolidated=bool(dataset.xarray_kwargs.get("consolidated"))
+        )
+
+        collection = xstac.xarray_to_stac(
+            ds,
+            template,
+            temporal_dimension=dataset.temporal_dimension or "time",
+            x_dimension=dataset.x_dimension or "lon",
+            y_dimension=dataset.y_dimension or "lat",
+            reference_system=dataset.reference_system or 4326,
+        )
+        return collection.to_dict()
+
+    def create_cog_collection(self, dataset: Dict[str, Any]) -> dict:
+        collection_stac = self.get_template(dataset)
+
+        # Override the extents if they exists
+        if (spatial_extent := dataset.get("spatial_extent")) or (
+            temporal_extent := dataset.get("temporal_extent")
+        ):
+            collection_stac["extent"] = SpatioTemporalExtent.parse_obj(
+                {
+                    "spatial": {"bbox": [list(spatial_extent)]},
+                    "temporal": {
+                        "interval": [
+                            # most of our data uses the Z suffix for UTC - isoformat() doesn't
+                            [
+                                x.isoformat().replace("+00:00", "Z")
+                                for x in list(temporal_extent)
+                            ]
+                        ]
+                    },
+                }
+            )
+
+        collection_stac["item_assets"] = {
```

Comment:
this isn't a required stac field so can only add it when provided in the request?

Reply:
I think we still want to include this as a default for `/dataset` inputs, since the generated defaults will result in a minimal dashboard-ready collection. For collections that deliberately have no `item_assets`, or have multiple assets, we should probably point users to the `/ingest/collections` endpoint.

Reply:
Personally, I don't think it is helpful to have a collection item_assets definition of cog_default in all collections by default but I don't think it poses much risk at this point when we do not have algorithmic parsing of collection metadata that would fail when collections falsely advertise cog_default assets. We do risk user confusion/mistrust of collection metadata so maybe we can revisit this on the next round.